22-discovery.ks

```diff
@@ -15,6 +15,9 @@ systemctl enable NetworkManager-wait-online.service
 echo " * enabling nm-prepare service"
 systemctl enable nm-prepare.service
 
+echo " * enabling nm-connected service"
+systemctl enable nm-connected.service
+
```

Comment:
You are right.
/usr/lib/systemd/system/network-online.target.wants/NetworkManager-wait-online.service comes from the NetworkManager package and is NOT provided by NetworkManager-wait-online.service [Install]-section. I remove `systemctl enable nm-connected.service` and give it a try.